nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -66,6 +66,30 @@ let
     #!${pkgs.runtimeShell}
     ${checkedSystemdBootBuilder} "$@"
     ${cfg.extraInstallCommands}
+    '';
+
+  loaderSettingsFormat = pkgs.formats.keyValue {
+    mkKeyValue = k: v: if v == null then "" else
+    lib.generators.mkKeyValueDefault { } " " k v;
+  };
+
+  loaderConfigFile = loaderSettingsFormat.generate "loader.conf" {
+    timeout = config.boot.loader.timeout;
+    console-mode = cfg.consoleMode;
+    editor = cfg.editor;
+    default = "nixos-*";
+  };
+
+  finalLanzabooteBuilder = pkgs.writeShellScript "install-lanzaboote.sh" ''
+    ${cfg.secureBoot.package}/bin/lzbt install \
+      --systemd ${config.systemd.package} \
+      --systemd-boot-loader-config ${loaderConfigFile} \
+      --public-key ${cfg.secureBoot.publicKeyFile} \
+      --private-key ${cfg.secureBoot.privateKeyFile} \
+      --configuration-limit ${toString (if cfg.configurationLimit == null then 0 else cfg.configurationLimit)} \
```

Comment:
Would it be possible to add a apply to set configurationLimit to 0 if it is null or remove null altogether instead?

Reply:
Out of scope, feel free to do the cleanup after this PR.